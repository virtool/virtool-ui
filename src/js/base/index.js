export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
