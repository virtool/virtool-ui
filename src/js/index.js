import * as Sentry from "@sentry/browser";
import { createBrowserHistory } from "history";
import "./nonce";
import "normalize.css";
import React from "react";
import ReactDOM from "react-dom";
import App from "./app/App";
import { createAppStore } from "./app/reducer";
import { Request } from "./app/request";
import { setInitialState } from "./app/actions";
if (module.hot) {
    module.hot.accept(err => {
        throw err;
    });
}
const sentryInit = res => {
    const dev = res.body.dev;
    if (!dev) {
        Sentry.init({
            dsn: "https://d9ea493cb0f34ad4a141da5506e6b03b@sentry.io/220541"
        });

        window.Sentry = Sentry;
    }
    return res;
};

window.captureException = error => (window.virtool.dev ? console.error(error) : window.Sentry.captureException(error));
const history = createBrowserHistory();
window.store = createAppStore(history);

Request.get("/api")
    .then(sentryInit)
    .then(({ body }) => window.store.dispatch(setInitialState(body)));

ReactDOM.render(<App store={window.store} history={history} />, document.getElementById("app-container"));
