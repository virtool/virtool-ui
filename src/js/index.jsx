import * as Sentry from "@sentry/browser";
import { createBrowserHistory } from "history";
import "./nonce";
import "normalize.css";
import React from "react";
import ReactDOM from "react-dom";
import App from "./app/App";
import { createAppStore } from "./app/reducer";
import { Request } from "./app/request";
import { setInitialState } from "./app/actions";

window.captureException = error => console.error(error);

if (window.__sentry_dsn__ !== "SENTRY_DSN") {
    window.captureException = error => Sentry.captureException(error);

    Sentry.init({
        dsn: window.__sentry_dsn__
    });
}

const history = createBrowserHistory();

window.b2c = { use: false };

window.store = createAppStore(history);

Request.get("/api").then(({ body }) => {
    window.store.dispatch(setInitialState(body));
});

ReactDOM.render(<App store={window.store} history={history} />, document.getElementById("app-container"));
