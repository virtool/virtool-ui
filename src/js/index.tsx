import * as Sentry from "@sentry/react";
import { withProfiler } from "@sentry/react";
import { createBrowserHistory } from "history";
import "normalize.css";
import React from "react";
import { createRoot } from "react-dom/client";
import App from "./app/App";
import "./nonce";

if (window.virtool.sentryDsn !== "SENTRY_DSN") {
    window.captureException = error => Sentry.captureException(error);

    Sentry.init({
        dsn: window.virtool.sentryDsn,
        integrations: [new Sentry.BrowserTracing()],
        tracesSampleRate: 0.5,
    });
} else {
    window.captureException = error => console.error(error);
}

const history = createBrowserHistory();

window.virtool.b2c = { use: false };

const AppWithProfiler = withProfiler(App);

const container = document.getElementById("app-container");
const root = createRoot(container);
root.render(<AppWithProfiler history={history} />);
