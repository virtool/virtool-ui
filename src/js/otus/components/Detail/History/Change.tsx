import { Attribution, BoxGroupSection, Icon, Label } from "@base";
import { useRevertOTU } from "@otus/queries";
import { OTUNested } from "@otus/types";
import { UserNested } from "@users/types";
import { get } from "lodash-es";
import React from "react";
import styled from "styled-components";

const methodIconProps = {
    add_isolate: {
        name: "flask",
        color: "blue",
    },
    create: {
        name: "plus-square",
        color: "blue",
    },
    create_sequence: {
        name: "dna",
        color: "blue",
    },
    edit: {
        name: "pencil-alt",
        color: "orange",
    },
    edit_isolate: {
        name: "flask",
        color: "orange",
    },
    edit_sequence: {
        name: "dna",
        color: "orange",
    },
    clone: {
        name: "clone",
        color: "blue",
    },
    import: {
        name: "file-import",
        color: "blue",
    },
    remote: {
        name: "link",
        color: "blue",
    },
    remove: {
        name: "trash",
        color: "red",
    },
    remove_isolate: {
        name: "flask",
        color: "red",
    },
    remove_sequence: {
        name: "dna",
        color: "red",
    },
    set_as_default: {
        name: "star",
        color: "orange",
    },
    update: {
        name: "arrow-alt-circle-up",
        color: "orange",
    },
};

function getMethodIcon(methodName: string) {
    const props = get(methodIconProps, methodName, {
        name: "exclamation-triangle",
        color: "red",
    });

    return <Icon {...props} />;
}

const StyledChange = styled(BoxGroupSection)`
    align-items: center;
    display: grid;
    grid-template-columns: 42px 2fr 1fr 15px;

    div:first-child {
        min-width: 42px;
    }
`;

const Description = styled.div`
    align-items: center;
    display: flex;

    i {
        margin-right: 5px;
    }
`;

type ChangeProps = {
    id: string;
    createdAt: string;
    description: string;
    methodName: string;
    otu: OTUNested;
    unbuilt: boolean;
    user: UserNested;
};

/**
 * A condensed change item for use in a list of changes
 */
export default function Change({ id, createdAt, description, methodName, otu, unbuilt, user }: ChangeProps) {
    const mutation = useRevertOTU(otu.id);

    return (
        <StyledChange>
            <div>
                <Label>{otu.version}</Label>
            </div>

            <Description>
                {getMethodIcon(methodName)}
                <span>{description || "No Description"}</span>
            </Description>

            <Attribution time={createdAt} user={user.handle} verb="" />

            {unbuilt && (
                <Icon
                    name="history"
                    tip="Revert"
                    onClick={() => (unbuilt ? mutation.mutate({ changeId: id }) : null)}
                />
            )}
        </StyledChange>
    );
}
