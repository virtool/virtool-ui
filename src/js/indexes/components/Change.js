import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { SpacedBox } from "../../base";

const StyledIndexChange = styled(SpacedBox)`
    display: grid;
    grid-template-columns: 1fr 1fr;
`;

export const IndexChange = ({ description, otuName }) => (
    <StyledIndexChange>
        <strong>{otuName}</strong>
        {description}
    </StyledIndexChange>
);

export const mapStateToProps = (state, ownProps) => {
    const { otu, description } = state.indexes.history.documents[ownProps.index];
    return {
        description,
        otuName: otu.name
    };
};

export default connect(mapStateToProps)(IndexChange);
