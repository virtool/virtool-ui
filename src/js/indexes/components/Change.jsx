import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { BoxSpaced } from "../../base";

const StyledIndexChange = styled(BoxSpaced)`
    display: grid;
    grid-template-columns: 1fr 1fr;
`;

export const IndexChange = ({ description, otuName }) => (
    <StyledIndexChange>
        <strong>{otuName}</strong>
        {description}
    </StyledIndexChange>
);

export const mapStateToProps = (state, ownProps) => {
    const { otu, description } = state.indexes.history.documents[ownProps.index];
    return {
        description,
        otuName: otu.name
    };
};

export default connect(mapStateToProps)(IndexChange);
