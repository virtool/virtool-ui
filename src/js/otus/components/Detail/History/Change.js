import { get } from "lodash-es";
import React, { useCallback } from "react";
import styled from "styled-components";

import { Attribution, BoxGroupSection, Icon, Label } from "../../../../base";

const methodIconProps = {
    add_isolate: {
        name: "flask",
        color: "blue"
    },
    create: {
        name: "plus-square",
        color: "blue"
    },
    create_sequence: {
        name: "dna",
        color: "blue"
    },
    edit: {
        name: "pencil-alt",
        color: "orange"
    },
    edit_isolate: {
        name: "flask",
        color: "orange"
    },
    edit_sequence: {
        name: "dna",
        color: "orange"
    },
    clone: {
        name: "clone",
        color: "blue"
    },
    import: {
        name: "file-import",
        color: "blue"
    },
    remote: {
        name: "link",
        color: "blue"
    },
    remove: {
        name: "trash",
        color: "red"
    },
    remove_isolate: {
        name: "flask",
        color: "red"
    },
    remove_sequence: {
        name: "dna",
        color: "red"
    },
    set_as_default: {
        name: "star",
        color: "orange"
    },
    update: {
        name: "arrow-alt-circle-up",
        color: "orange"
    }
};

const getMethodIcon = methodName => {
    const props = get(methodIconProps, methodName, {
        name: "exclamation-triangle",
        color: "red"
    });

    return <Icon {...props} />;
};

const StyledChange = styled(BoxGroupSection)`
    align-items: center;
    display: grid;
    grid-template-columns: 42px 2fr 1fr 15px;

    div:first-child {
        min-width: 42px;
    }
`;

const Description = styled.div`
    align-items: center;
    display: flex;

    i {
        margin-right: 5px;
    }
`;

export const Change = ({ id, createdAt, description, methodName, otu, unbuilt, user, onRevert }) => {
    const handleRevert = useCallback(() => {
        onRevert(otu.id, otu.version, id);
    }, [otu.id, otu.version, id]);

    return (
        <StyledChange>
            <div>
                <Label>{otu.version}</Label>
            </div>

            <Description>
                {getMethodIcon(methodName)}
                <span>{description || "No Description"}</span>
            </Description>

            <Attribution time={createdAt} user={user.id} verb="" />

            {unbuilt && <Icon name="history" tip="Revert" onClick={handleRevert} />}
        </StyledChange>
    );
};
