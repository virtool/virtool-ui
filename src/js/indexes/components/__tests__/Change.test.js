import { IndexChange, mapStateToProps } from "../Change.js";

describe("<IndexChange />", () => {
    let props;

    beforeEach(() => {
        props = {
            description: "foo",
            otuName: "bar"
        };
    });

    it("should render", () => {
        const wrapper = shallow(<IndexChange {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [otuName=null]", () => {
        props.otuName = null;
        const wrapper = shallow(<IndexChange {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps", () => {
    it("should return props", () => {
        const state = {
            indexes: {
                history: {
                    documents: [
                        {
                            otu: {
                                name: "Baz"
                            },
                            description: "Boom"
                        }
                    ]
                }
            }
        };
        const ownProps = {
            index: 0
        };
        const props = mapStateToProps(state, ownProps);

        expect(props).toEqual({
            description: "Boom",
            otuName: "Baz"
        });
    });
});
