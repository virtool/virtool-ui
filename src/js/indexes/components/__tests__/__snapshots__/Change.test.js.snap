// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexChange /> should render 1`] = `
<Change__StyledIndexChange>
  <strong>
    bar
  </strong>
  foo
</Change__StyledIndexChange>
`;

exports[`<IndexChange /> should render when [otuName=null] 1`] = `
<Change__StyledIndexChange>
  <strong />
  foo
</Change__StyledIndexChange>
`;
